pkg/tests/api/alerting/api_notification_channel_test.go

```diff
@@ -811,8 +811,9 @@ func TestNotificationChannels(t *testing.T) {
 	// Eventually, we'll get all the desired alerts.
 	// nolint:gosec
 	require.Eventually(t, func() bool {
-		// TODO: not waiting for the failed notifications, flaky test?
-		return mockChannel.totalNotifications() >= len(nonEmailAlertNames) && len(mockEmail.emails) >= 1
+		return mockChannel.totalNotifications() >= len(nonEmailAlertNames) &&
+			mockChannel.totalNotificationErrors() >= len(expNotificationErrors) &&
```

Comment:
Yes, but this shouldn't be the case since I'm forcing notifications to fail from the server side.
If a receiver is marked as a "failing" receiver, it will fail. It doesn't depend on other conditions.